genai-perf/tests/test_cli.py

```diff
@@ -917,3 +917,75 @@ def test_get_extra_inputs_as_dict(self, extra_inputs_list, expected_dict):
         namespace.extra_inputs = extra_inputs_list
         actual_dict = parser.get_extra_inputs_as_dict(namespace)
         assert actual_dict == expected_dict
+
+    TEST_TRITON_METRICS_URL = "http://custom-metrics-url:8002/metrics"
+    INVALID_URL = "invalid_url"
+    INVALID_URL_ERROR_MESSAGE = (
+        "The URL passed for --server-metrics-url is invalid. "
+        "It must use 'http' or 'https', have a valid domain, "
+        "and contain '/metrics' in the path. The expected structure is: "
+        "<scheme>://<netloc>/<path>;<params>?<query>#<fragment>"
+    )
+
+    @pytest.mark.parametrize(
+        "args_list, expected_url, expected_error",
+        [
+            # Test with a custom URL
+            (
+                [
+                    "genai-perf",
+                    "profile",
+                    "--model",
+                    "test_model",
+                    "--service-kind",
+                    "triton",
+                    "--server-metrics-url",
+                    TEST_TRITON_METRICS_URL,
+                ],
+                TEST_TRITON_METRICS_URL,
+                None,
+            ),
+            # Test with default URL
+            (
+                [
+                    "genai-perf",
+                    "profile",
+                    "--model",
+                    "test_model",
+                    "--service-kind",
+                    "triton",
+                ],
+                None,
+                None,
+            ),
+            # Test with invalid URL
+            (
+                [
+                    "genai-perf",
+                    "profile",
+                    "--model",
+                    "test_model",
+                    "--service-kind",
+                    "triton",
+                    "--server-metrics-url",
+                    INVALID_URL,
+                ],
+                None,
+                INVALID_URL_ERROR_MESSAGE,
+            ),
+        ],
+    )
+    def test_server_metrics_url_for_triton(
```

Comment:
do we have any testing for properly formed urls that are not actually a reachable endpoint?

Reply:
Yes, there are tests in test_telemetry_data_collector.py that uses valid URLs. There's one test for success scenario and one test for all kinds of failure scenarios like 404, 403, 500, etc. These are already merged to main. 

This is the URL 
` TEST_SERVER_URL = "http://testserver:8080/metrics"`

This is the test
![image](https://github.com/user-attachments/assets/30945da6-6420-417a-82f0-c4ccfd48990a)

Reply:
perfect. thank you for catching me up here.